include/exadg/operators/operator_base.cpp

```diff
@@ -759,12 +761,24 @@ template<int dim, typename Number, int n_components>
 template<typename SparseMatrix>
 void
 OperatorBase<dim, Number, n_components>::internal_init_system_matrix(
-  SparseMatrix & system_matrix) const
+  SparseMatrix &   system_matrix,
+  MPI_Comm const & mpi_comm) const
 {
   DoFHandler<dim> const & dof_handler = this->matrix_free->get_dof_handler(this->data.dof_index);
 
   IndexSet const & owned_dofs =
     is_mg ? dof_handler.locally_owned_mg_dofs(this->level) : dof_handler.locally_owned_dofs();
+
+  // check for a valid subcommunicator - the second check is needed on those
+  // MPI ranks which will not participate in the actual communication, i.e.,
+  // which are left out
+  AssertThrow(Utilities::MPI::sum(owned_dofs.n_elements(), mpi_comm) == owned_dofs.size() ||
```

Comment:
Question: does `owned_dofs.size()` involve a call to MPI, or why is it more than the dofs of one MPI process? Should be documented more clearly in my opinion.

Reply:
One is the number of dofs on the present process, one is for the global sum. I have the check because a subcommunicator leaves out some ranks and we need to make sure that none of those carries dofs. Fixed in cc655075